src/puppetlabs/puppetdb/query/aggregate_event_counts.clj

```diff
@@ -31,20 +31,22 @@
                                (event-counts/query->sql
                                  true
                                  version
-                                 query [summarize_by query-options {}]))]
+                                 query
+                                 query-options))]
     (vector (get-aggregate-sql count-sql summarize_by) params)))
 
 (defn query->sql
   "Convert an aggregate-event-counts `query` and a value(s) to `summarize_by`
    into a SQL string. Since all inputs are forwarded to
    `event-counts/query->sql`, look there for proper documentation."
-  [version query [summarize_by {:keys [distinct_resources?] :as query-options}]]
+  [version query {:keys [distinct_resources? summarize_by] :as query-options}]
   {:pre  [((some-fn nil? sequential?) query)
           ((some-fn map? nil?) query-options)]
    :post [(jdbc/valid-jdbc-query? (:results-query %))]}
   (let [summary-vec (str/split summarize_by #",")
         nsummarized (count summary-vec)
-        aggregate-fn #(assemble-aggregate-sql version query % query-options)
+        aggregate-fn #(assemble-aggregate-sql version query (assoc query-options
+                                                                   :summarize_by %))
```

Comment:
Maybe `#(->> % (assoc query-options :summarize_by) (assemble-aggregate-sql version query))` to make this a little more readable